dbms/src/Operators/tests/gtest_auto_pass_through_helper.cpp

```diff
@@ -0,0 +1,146 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+#include <AggregateFunctions/registerAggregateFunctions.h>
+#include <Core/ColumnWithTypeAndName.h>
+#include <DataTypes/DataTypeDecimal.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Interpreters/Context.h>
+#include <Operators/AutoPassThroughHashAggHelper.h>
+#include <TestUtils/ColumnGenerator.h>
+#include <TestUtils/TiFlashTestBasic.h>
+#include <TestUtils/TiFlashTestEnv.h>
+#include <gtest/gtest.h>
+
+namespace DB
+{
+namespace tests
+{
+
+class TestAutoPassThroughHelper : public ::testing::Test
```

Comment:
IMO, we need to cover agg functions not only sum, min/max, first_row, since other agg functions  will be affected also.

Reply:
The test framework doesn't support other agg funcs for now. And i will add an endless integration case later, so I didn't test here.
check https://github.com/pingcap/tiflash/blob/954f1477018a0fdf87d7106ce466a16a7fdc82c1/dbms/src/Debug/MockExecutor/AggregationBinder.cpp#L283-L287